event/eth_sign_redeem.go

```diff
@@ -121,21 +121,30 @@ func (j *JobETHSignRedeem) DoMyJob(ctx interface{}) {
 	status, err := cd.VerifyRedeem(addr, msg.From())
 	if err != nil {
 		ethCtx.Logger.Error("Error in verifying redeem :", j.GetJobID(), err)
-		return
 	}
+	//Ethereum connectivity issue
+	if status == -2 {
+		ethCtx.Logger.Error("Unable to connect to ethereum smartcontract")
+		j.Status = jobs.Failed
+		BroadcastReportFinalityETHTx(ctx.(*JobsContext), j.TrackerName, j.JobID, false)
+	}
+	//Signature confirmed
 	if success {
 		ethCtx.Logger.Debug("validator Sign Confirmed | validator Address (SIGNER):", ethCtx.GetValidatorETHAddress().Hex(), "| User Eth Address :", msg.From().Hex())
 		j.Status = jobs.Completed
 		return
 	}
+
 	if j.RetryCount >= 0 && !success {
```

Comment:
No point in checking !success.. as it will always be true at this place. You can try to move this condition above , if that makes sense

Reply:
Or probably check for err != nil